pkg/v1/webhooks/certificates.go

```diff
@@ -63,50 +63,64 @@ func addCertsToWebhookConfigs(ctx context.Context, k8sclient kubernetes.Interfac
 	return nil
 }
 
-func NewTLSSecret(ctx context.Context, secretName, serviceName, certPath, keyPath, namespace string) (*corev1.Secret, error) {
-	secret, err := resources.MakeSecretInternal(ctx, secretName, namespace, serviceName)
-	if err != nil {
-		return nil, err
-	}
+// func NewTLSSecret(ctx context.Context, secretName, serviceName, certPath, keyPath, namespace string) (*corev1.Secret, error) {
+//	secret, err := resources.MakeSecret(ctx, secretName, namespace, serviceName)
+//	if err != nil {
+//		return nil, err
+//	}
+
+//	return secret, nil
+//}
 
+func WriteServerTLSToFileSystem(ctx context.Context, certPath, keyPath string, secret *corev1.Secret) error {
```

Comment:
Probably a naive question: why do we need to write the cert to a secret in addition to writing to file system? IIRC, cert-manager creates a secret which you can mount as volume to the controller pod. But we're in full control of cert generation here and we're already writing it to file system (which controller-runtime keeps a watch on).

Reply:
You are correct, we could just keep it in memory and write to the file system, (which btw was our original implementation during our POC) but 
the filesystem is ephemeral, so if the kapp-controller is restarted the certificates would be lost. 
we write the certificates to the secret so we can use the same certificate continously accross restarts. 
and also so it is available from the outside world. 
Theoretically (although not supported) you can upload your own certificates to the secret.

Reply:
Ah, that makes sense. Thanks!